gordon/resources/s3.py

```diff
@@ -265,13 +265,13 @@ def __init__(self, *args, **kwargs):
 
     def get_bucket_arn(self):
         bucket = self.settings.get('bucket')
-        if bucket.startswith('arn:aws:'):
+        if isinstance(bucket, six.string_types) and bucket.startswith('arn:aws:'):
             return bucket
         return troposphere.Join("", ["arn:aws:s3:::", bucket])
```

Comment:
What will happen when bucket is a Ref and it is then joined with a string?

This method might need to detect it is dealing with a Ref then just return the Ref instance unchanged.